crates/blockchain/fork_choice.rs

```diff
@@ -134,6 +134,7 @@ pub fn apply_fork_choice(
         store.update_safe_block_number(safe.header.number)?;
     }
     store.update_latest_block_number(head.number)?;
+    store.update_sync_status(true)?;
```

Comment:
Are we sure this is the way to determine that the sync is finished? We can receive a fork choice updated engine message pretty fast down the line even if we didn't sync yet.

Reply:
That's the point, if we receive a fork choice and we are not synced yet then we would return a `Syncing` response and continue to the sync, we would not apply the fork choice so this is the correct place to check wether we are currently synced or not